framework/src/geomsearch/PenetrationLocator.C

```diff
@@ -127,7 +127,13 @@ PenetrationLocator::detectPenetration()
 void
 PenetrationLocator::reinit()
 {
+  // Delete the PenetrationInfo objects we own before clearing the
+  // map, or we have a memory leak.
+  for (auto & it : _penetration_info)
+    delete it.second;
```

Comment:
(yes, I'm sure I've written a ton of classes with data members which weren't encapsulated but should have been - it *is* possible to be a professional 80% of the time and a madman the other 20%)